Allow settings TLS protocols in WebSocket options

noman720 opened the pull request:
For android API level < 20, TLSv1.2 doesn't support by default. To enable TLSv1.1 and TLSv1.2 we need to enable TLS on socket.

How to use:
```
WebSocketConnection webSocketConnection = new WebSocketConnection();
WebSocketOptions options = new WebSocketOptions();
options.setEnableTls(true);
webSocketConnection.connect(SERVER_URL, handler, options);
```

oberstet commented:
First off: thanks for contributing to this project!

rgd the specific code, I have 2 issues;) 

* https://github.com/crossbario/autobahn-java/issues/374#issuecomment-374902150
* missing docstrings (this is public API, hence must have docstrings)

If you agree, and adjust the PR, we'll merge it right away.

noman720 commented:
Thanks for the guide.
- Added docstring
- Modified the implementation to give flexibility on setting protocols by user

**How to use**
```
WebSocketConnection webSocketConnection = new WebSocketConnection();
WebSocketOptions options = new WebSocketOptions();
options.setTLSEnabledProtocols(new String[]{"TLSv1.1", "TLSv1.2"});
webSocketConnection.connect(SERVER_URL, handler, options);
```

oberstet commented:
Thanks for the patch!
